### PR TITLE
Add Kubeflow and MLflow demos

### PR DESCRIPTION
**Pull request by chanchiwai-ray**
This is a WIP.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,36 +81,6 @@ juju controllers
 juju clouds
 ```
 
-### Deploy kubeflow
```
Comment (jneo8) [inline README.md:82]: Remove deploy kubeflow from here will make this document incomplete.

Suggest to keep it in both side or create a independent document for deploy kubeflow. It depends on we are creating two series or just one.

File: demos/kubeflow/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+$ juju relate istio-pilot mlflow-server
+$ juju relate mlflow-db mlflow-server
+$ juju relate mlflow-server admission-webhook
+```
+
+You can go to the mlflow dashboard using the IP and port given by the following
+commands.
+
+```sh
+$ microk8s kubectl get svc mlflow-server -nkubeflow
```
Comment (jneo8) [inline demos/kubeflow/README.md:53]: typo

File: demos/kubeflow/README.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+The kubeflow dashboard will available at `$IPADDR.nip.io`
+
+To run this demo, we will also need mlflow.
+
+```sh
+$ juju deploy mlflow-server
+$ juju deploy charmed-osm-mariadb-k8s mlflow-db
+$ juju relate minio mlflow-server
```
Comment (jneo8) [inline demos/kubeflow/README.md:43]: Reader for this document  may not familiar with juju.
We can add some command for them to verify the change or provide some comment. e.g.

* What is juju relate? ...

```sh
juju status -m model_name --relations
```

File: demos/kubeflow/README.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+$ juju config minio access-key=minio
+$ juju config minio secret-key=minio123
+```
+
+Test your access to Minio by accessing the dashboard. You can change the service
+to be of LoadBalancer type or use port-forwarding to access it. To access the
+minio dashboard, you find the IP and port using:
+
+```sh
+$ microk8s kubectl get svc minio -nkubeflow
```
Comment (jneo8) [inline demos/kubeflow/README.md:77]: typo

File: demos/kubeflow/README.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+The pipeline expects buckets `mlflow` and `mlpipeline` to exist in Minio. You
+can create those manually in the Minio UI if it's not created for you.
```
Comment (jneo8) [inline demos/kubeflow/README.md:81]: So is the buckets be created automatically?

File: demos/kubeflow/README.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+MLFlow manually.  Apply the files `k8s-files/allow-minio.yaml` and
+`k8s-files/allow-mlflow.yaml` to your user namespace.
+
+For example, if the user is the "admin" namespace:
+
+```sh
+$ microk8s kubectl apply -f k8s-files/allow-minio.yaml -n admin
+$ microk8s kubectl apply -f k8s-files/allow-mlflow.yaml -n admin
+```
+
```
Comment (jneo8) [inline demos/kubeflow/README.md:96]: Maybe some end wording

File: demos/kubeflow/k8s-files/allow-minio.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+apiVersion: kubeflow.org/v1alpha1
```
Comment (jneo8) [inline demos/kubeflow/k8s-files/allow-minio.yaml:1]: The naming of file can be improved if it told us which kind of resource it is.

File: demos/kubeflow/k8s-files/allow-mlflow.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+apiVersion: kubeflow.org/v1alpha1
```
Comment (jneo8) [inline demos/kubeflow/k8s-files/allow-mlflow.yaml:1]: The naming of file can be improved if it told us which kind of resource it is.

File: demos/kubeflow/README.md (hunk 2 above)
Comment (chanchiwai-ray) [reply demos/kubeflow/README.md:53]: it's actuall not a typo if you're saying `-nkubelfow`, it's valid and equals to `-n kubeflow`

File: demos/kubeflow/README.md (hunk 4 above)
Comment (chanchiwai-ray) [reply demos/kubeflow/README.md:77]: same above

**Pull request merged**